pkg/filter/plugins/plugins.go

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package plugins
+
+import (
+	"errors"
+	"plugin"
+	"strings"
+)
+
+import (
+	"github.com/dubbogo/dubbo-go-proxy/pkg/common/constant"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/config"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/context"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/filter"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/logger"
+)
+
+var (
+	apiURLWithPluginsMap = make(map[string]context.FilterChain)
+	groupWithPluginsMap  = make(map[string]map[string]PluginsWithFunc)
+	errEmptyPluginConfig = errors.New("Empty plugin config")
+)
+
+// PluginsWithFunc is a single plugin details
+type PluginsWithFunc struct {
+	Name     string
+	Priority int
+	fn       context.FilterFunc
+}
+
+// InitPluginsGroup prase api_config.yaml(pluginsGroup) to map[string][]PluginsWithFunc
+func InitPluginsGroup(groups []config.PluginsGroup, filePath string) {
+
+	if "" == filePath || len(groups) == 0 {
+		return
+	}
+
+	// load file.so
+	pls, err := plugin.Open(filePath)
```

Comment:
configs/plugins/plugin.so should not be included in the project beacuse it's a bin file. Maybe build it in workflows is a better way.